### PR TITLE
Support local update file for OTA

### PR DESCRIPTION
**Pull request by agners**
Support local update file for Matter OTA updates. The otaUrl must use the file:/// URL scheme. The path is relative to the OTA provider directory.

### CODE REVIEW

**Review by marcelveldt — APPROVED** [review]

File: matter_server/server/ota/provider.py [hunk 1]
```diff
@@ -326,6 +302,55 @@ async def download_update(self, update_desc: dict) -> None:
...
+        else:
+            LOGGER.warning(
+                "No OTA checksum type or not supported, OTA will not be checked."
+            )
+
+        if parsed_url.scheme in ["http", "https"]:
+            file_path = await self._download_update(url, checksum_alg)
+        elif parsed_url.scheme in ["file"]:
+            file_path = self._ota_provider_base_dir / Path(parsed_url.path[1:])
+            if not file_path.exists():
```
Comment (MartinHjelmare) [inline matter_server/server/ota/provider.py:331]: This is a blocking syscall.

File: matter_server/server/ota/provider.py [hunk 2]
```diff
@@ -326,6 +302,55 @@ async def download_update(self, update_desc: dict) -> None:
...
+                logging.warning("Local update file not found: %s", file_path)
+                raise UpdateError("Local update file not found")
+            if checksum_alg:
+                checksum_alg.update(
+                    await loop.run_in_executor(None, file_path.read_bytes)
```
Comment (MartinHjelmare) [inline matter_server/server/ota/provider.py:336]: We can use `asyncio.to_thread` now. It does the same thing.

File: matter_server/server/ota/provider.py (hunk 1 above)
Comment (agners) [reply matter_server/server/ota/provider.py:331]: Implemented with #886.

File: matter_server/server/ota/provider.py (hunk 2 above)
Comment (agners) [reply matter_server/server/ota/provider.py:336]: Implemented with #886.

**Pull request merged**